compiler/noirc_frontend/src/hir/type_check/errors.rs

```diff
@@ -511,6 +513,13 @@ impl<'a> From<&'a TypeCheckError> for Diagnostic {
             TypeCheckError::CyclicType { typ: _, span } => {
                 Diagnostic::simple_error(error.to_string(), "Cyclic types have unlimited size and are prohibited in Noir".into(), *span)
             }
+            TypeCheckError::CannotInvokeStructFieldFunctionType { method_name, object_type, span } => {
+                Diagnostic::simple_error(
+                    format!("No method named '{method_name}' found for type '{object_type}'"), 
+                    format!("to call the function stored in '{method_name}', surround the field access with parentheses: '(', ')'"),
```

Comment:
The example is good but I can also see it confusing users if their object isn't `self`

Reply:
We can't assume that it's being called on `self` but otherwise yes.

Reply:
True, there is a `has_self_arg` we can use, but I guess it could be a variable, and if it's `false` I don't know what to put that isn't even more confusing. Okay never mind.